tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java

```diff
@@ -0,0 +1,319 @@
+
+        // insert before existing headers header
+        for (String header : HEADERS_AFTER_BUNDLE_VEIRSION) {
+            // add Bundle-Version before
+            if (localtion.contains(header)) {
+                int versionHeaderStartIndex = localtion.indexOf(header);
+                if (wrapProtocolOptions.contains("$")) {
+                    // "amp;" is automatically added
+                    return sb.insert(versionHeaderStartIndex, "%s&".formatted(bundleVersionHeader)).toString();
+                } else {
+                    // "amp;" is automatically added
+                    return sb.insert(versionHeaderStartIndex, "$%s&".formatted(bundleVersionHeader)).toString();
+                }
+            }
+        }
+
+        // insert at the end
+        if (wrapProtocolOptions.contains("$")) {
+            // "amp;" is automatically added
+            return sb.insert(localtion.length(), "&%s".formatted(bundleVersionHeader)).toString();
+        } else {
+            return sb.insert(localtion.length(), "$%s".formatted(bundleVersionHeader)).toString();
+        }
+    }
+
+    /**
+     * @param location
+     * @return artifact version first char index, inclusive
+     */
+    int getVersionStartIndex(String location) {
+        char[] chars = location.toCharArray();
+
+        boolean artifactIdFound = false;
+        for (int i = 0; i < chars.length; i++) {
+            if ('/' == chars[i]) {
+                if (!artifactIdFound) {
+                    artifactIdFound = true;
+                } else {
+                    return i + 1;
+                }
+            }
+        }
+
+        return -1;
+    }
+
+    int getVersionEndIndex(String location) {
+        return getVersionEndIndex(location, getVersionStartIndex(location));
+    }
+
+    /**
+     * @param location
+     * @param versionStartIndex
+     * @return artifact version last char index, inclusive
+     */
+    int getVersionEndIndex(String location, int versionStartIndex) {
+        char[] chars = location.toCharArray();
+
+        // start at + 1 to ignore the potential $ coming from version placeholder
+        for (int i = versionStartIndex + 1; i < chars.length; i++) {
+            if ('$' == chars[i]) {
+                return i - 1;
+            }
+        }
+
+        return chars.length - 1;
+    }
+
+    String getVersion(String Location) {
+        return getVersion(Location, getVersionStartIndex(Location), getVersionEndIndex(Location));
+    }
+
+    String getVersion(String location, int versionStartIndex, int versionEndIndex) {
+        return location.substring(versionStartIndex, versionEndIndex + 1);
+    }
+
+    String getValidVersion(String location, String version) throws Exception {
+        if (version.charAt(0) == '$') {
+            throw new Exception("Maven version placeholder '%s' wasn't resolved".formatted(version));
+        }
+
+        try {
+            // Test if version will work in Karaf
+            new Version(version);
```

Comment:
Why do we need this step? why not call directly `VersionCleaner.clean(version)`? in other words, we could do directly what we have in the catch block, or maybe I miss something?

Reply:
I was worried about having an already correct Bundle-Version, like `33.2.0.jre`, but the output of `VersionCleaner.clean(33.2.0.jre)` returning something different. I don't have a strong opinion about this since I can't find any example...

Reply:
I've found it! 

Given the input:
`wrap:mvn:com.github.java-json-tools/json-patch/1.13`
What I had before generated:
`wrap:mvn:com.github.java-json-tools/json-patch/1.13$Bundle-Version=1.13`
What I have now generates (after suggested changes):
`wrap:mvn:com.github.java-json-tools/json-patch/1.13.0$Bundle-Version=1.13.0`

So inside Karf, we get 
`114 | Active |  50 | 1.13.0         | json-patch`
instead of 
`114 | Active |  50 | 1.13           | json-patch`